src/record_syscall.cc

```diff
@@ -2130,7 +2130,13 @@ static Switchable prepare_ptrace(RecordTask* t,
     case PTRACE_SYSEMU_SINGLESTEP:
     case PTRACE_CONT: {
       RecordTask* tracee = verify_ptrace_target(t, syscall_state, pid);
+      // If the tracer wants to observe syscall entries, we can't use the
+      // syscallbuf, because the tracer may want to change syscall numbers
+      // which the syscallbuf code is not prepared to handle.
       if (tracee) {
+        tracee->set_syscallbuf_locked(command == PTRACE_SYSCALL ||
+                                      command == PTRACE_SYSEMU ||
+                                      command == PTRACE_SYSEMU_SINGLESTEP);
```

Comment:
If we're going to do this for `PTRACE_CONT` then we're effectively doing it for all tracees and I think it's simpler and more robust to do this as soon as we attach.

Then again, do we need to do this for all tracees? Maybe it's more robust if we do, but that would mean disabling syscallbuf for any tracee with any kind of ptrace supervisor (including Wine?) so that might be bad.

So I lean towards not doing this for `PTRACE_CONT` unless you make the case that we should.